galley/pkg/config/schema/ast/ast.go

```diff
@@ -21,12 +21,15 @@ import (
 	"github.com/ghodss/yaml"
 )
 
+// Direct transform's name. Used for parsing.
+const Direct = "direct"
+
 // Metadata is the top-level container.
 type Metadata struct {
-	Collections []*Collection `json:"collections"`
-	Snapshots   []*Snapshot   `json:"snapshots"`
-	Sources     []Source      `json:"sources"`
-	Transforms  []Transform   `json:"transforms"`
+	Collections       []*Collection       `json:"collections"`
+	Snapshots         []*Snapshot         `json:"snapshots"`
```

Comment:
Should we rename these to "SnapshotGroups"? We already use that terminology in conversation and a lot of the code, and it will probably be helpful to consistently distinguish between a snapshot (instance) and snapshotgroup (definition). We could also go with SnapshotSettings to be consistent with TransformSettings. (If so, we should probably update comments & variables that refer to snapshotgroups accordingly).

Reply:
I think I'll keep these as Snapshots. Given these are metadata (based on the package and the config contents), there is not much risk of confusion between an instance and a definition. Also, we use the term snapshotGroup internally in the runtime for the ephemeral state.


The reason I am changing the name from Transform to TransformSettings is to be able to introduce other transform related objects into the schema pacakge (.e.g TransformInfo, to capture input/output relationsships for example).